color_generator/models/color_model.py

```diff
@@ -1,33 +1,30 @@
 """ColorModel class."""
-from typing import Callable, Dict, Tuple
-import numpy as np
+from os import truncate
+from color_generator.datasets.colors_dataset import ColorsDataset
 
 from color_generator.models.base import Model
-from color_generator.datasets.colors_dataset import ColorsDataset
-from color_generator.networks.distilbert import distilbert
+from color_generator.datasets.dataloaders import DataLoaders
+from color_generator.networks.distilbert import Distilbert
 
 from transformers import DistilBertTokenizer
-from tensorflow import convert_to_tensor
 
 
 class ColorModel(Model):
     """ColorModel works on datasets providing single text sequence up to 32 tokens, with RGB labels."""
 
-    def __init__(
-        self,
-        dataset_cls: type = ColorsDataset,
-        network_fn: Callable = distilbert,
-        dataset_args: Dict = None,
-        network_args: Dict = None,
-    ):
-        super().__init__(dataset_cls, network_fn, dataset_args, network_args)
-
-    def predict_on_text(self, input_text: str) -> np.array:
-        tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-uncased')
-        input_text = tokenizer(input_text, max_length=32)
-        predictions = self.network.predict({
-            'input_ids': convert_to_tensor([input_text['input_ids']], np.int32),
-            'attention_mask': convert_to_tensor([input_text['attention_mask']], np.int32)
-        })
-
-        return predictions
+    def __init__(self, dataloaders, network_fn, device):
+        super().__init__(dataloaders, network_fn, device)
+
+    def predict_on_text(self, input_text):
+        tokenizer = DistilBertTokenizer.from_pretrained("distilbert-base-uncased")
```

Comment:
Same as in color_datasets you could use AutoTokenizer and parameterize model name